.github/CONTRIBUTING.md

```diff
@@ -212,7 +212,20 @@ invoke `make format`.
 The target `make formatall` in the root will automatically run all style checks
 and make any required changes. PRs must pass the formatting checks before landing.
 
-### Step 8: Landing
+### Step 8: Licensing
+
+Tock is dual-licensed under Apache2/MIT. Files in Tock require a short header
+indicating this. In 99% of cases, the following text should be at the top of every file:
+
+> Copyright The Tock Project Authors.
```

Comment:
Do we want to allow an optional year here? i.e.
```
Copyright [optional YYYY] The Tock Project Authors
```

Then we could leave off the year for existing files (which is difficult to determine in bulk), and new files can have accurate years.

Reply:
The same maintenance question applies. Is that the year the file was first committed or the year of the last update to it?

My real opinion though remains: if this is necessary for collaborators, then we should do it, otherwise keep the bare minimum.

Reply:
Okay, so from my Google-supported* armchair lawyering: in the US we would likely be fine without a year [it would default to the time of publication; adding a year would allow for subsequent renewal if there were substantial updates, but we don't really care about that]. Once you start looking internationally, however, the year is sometimes still significant.

Relatedly, putting a year _too early_ can invalidate a copyright, but putting one later than publication is okay (with the understanding that _technically_ if someone could produce evidence of their own published copy of the work prior to the date you declared, they could then invalidate your copyright; not something I think we are concerned with. In our case, courtesy of GitHub we have strong evidence of the first publication date of all of our work, should we ever need to get in such a fight).

Bonus things learned, while simply the word "Copyright" is sufficient, it is strongly recommended to also include ©, as some courts [e.g. Canada apparently](https://www.copyrightlaws.com/copyright-symbol-notice-year/) will give extra weight to the ©.

I will update this PR to try to capture these things.

For the initial bulk update, I believe we should mark everything as © 2020.

*The colloquial me-as-person-who-searched-things support, not any formal support from Google the corporate entity

Reply:
Did something suggest that we should be putting a copyright range? I don't know if I've ever seen "Copyright 2020-2021" or similar in a file before. It looks to me as though it would imply that there is not a copyright after that date...

Reply:
> Did something suggest that we should be putting a copyright range? I don't know if I've ever seen "Copyright 2020-2021" or similar in a file before. It looks to me as though it would imply that there is not a copyright after that date...

FWIW I've seen such year ranges quite often. But IANAL in terms of what putting a year or not and which year to put implies in terms of copyright.

Reply:
Re range: yes. I thought I explained that in the text I wrote:
```
If you create a new "version or edition" of a given work, then the date range
should be extended to include the newest work. In practice, this means any
update to a source file will require updating the range (and some churn
hopefully largely restricted to January each year).
```
Happy to take language updates to make that clearer.